src/components/organisms/EntityDetails/EntityDetailsContent/EntityDetailsContent.tsx

```diff
@@ -74,6 +76,7 @@ const EntityDetailsContent: React.FC = () => {
   const testIcon = entityDetails?.testIcon;
 
   const onRunButtonClick = async () => {
```

Comment:
Shouldn't it instead set the loading status based on `runEntity` result in `onRunButtonClick`?

Reply:
there's the possibility to access the loading state of the request using the "isLoading" flag provided directly by the RTK mutation. tried this first, but that's not enough, as the request is resolved but the agent is still running the test. If there are too many tests running in parallel, the agent will stop responding and return 503s. the agreement was, to have a "fixed" 2s window for this. The issue is impacting cloud especially.